functions.php

```diff
@@ -317,88 +323,91 @@ function wp_autoupdates_plugins_bulk_actions( $actions ) {
+		return;
+	}
 
-		if ( is_multisite() && ! is_network_admin() ) {
-			wp_die( __( 'Please connect to your network admin to manage plugins automatic updates.', 'wp-autoupdates' ) );
-		}
+	check_admin_referer( 'updates' );
 
-		$plugin = ! empty( esc_html( $_GET['plugin'] ) ) ? wp_unslash( esc_html( $_GET['plugin'] ) ) : '';
-		$page   = isset( $_GET['paged'] ) && ! empty( esc_html( $_GET['paged'] ) ) ? wp_unslash( esc_html( $_GET['paged'] ) ) : '';
-		$status = isset( $_GET['plugin_status'] ) && ! empty( esc_html( $_GET['plugin_status'] ) ) ? wp_unslash( esc_html( $_GET['plugin_status'] ) ) : '';
-		$s      = isset( $_GET['s'] ) && ! empty( esc_html( $_GET['s'] ) ) ? wp_unslash( esc_html( $_GET['s'] ) ) : '';
+	if ( ! current_user_can( 'update_plugins' ) || ! wp_autoupdates_is_plugins_auto_update_enabled() ) {
+		wp_die( __( 'Sorry, you are not allowed to enable plugins automatic updates.', 'wp-autoupdates' ) );
+	}
 
-		if ( empty( $plugin ) ) {
-			wp_redirect( self_admin_url( "plugins.php?plugin_status=$status&paged=$page&s=$s" ) );
-			exit;
-		}
+	if ( is_multisite() && ! is_network_admin() ) {
+		wp_die( __( 'Please connect to your network admin to manage plugins automatic updates.', 'wp-autoupdates' ) );
+	}
 
-		check_admin_referer( 'updates' );
+	$plugin = ! empty( esc_html( $_GET['plugin'] ) ) ? wp_unslash( esc_html( $_GET['plugin'] ) ) : '';
+	$page   = isset( $_GET['paged'] ) && ! empty( esc_html( $_GET['paged'] ) ) ? wp_unslash( esc_html( $_GET['paged'] ) ) : '';
+	$status = isset( $_GET['plugin_status'] ) && ! empty( esc_html( $_GET['plugin_status'] ) ) ? wp_unslash( esc_html( $_GET['plugin_status'] ) ) : '';
+	$s      = isset( $_GET['s'] ) && ! empty( esc_html( $_GET['s'] ) ) ? wp_unslash( esc_html( $_GET['s'] ) ) : '';
 
-		$wp_auto_update_plugins = get_site_option( 'wp_auto_update_plugins', array() );
-
-		if ( in_array( $plugin, $wp_auto_update_plugins, true ) ) {
-			$wp_auto_update_plugins = array_diff( $wp_auto_update_plugins, array( $plugin ) );
-			$action_type            = 'disable-autoupdate=true';
-		} else {
-			array_push( $wp_auto_update_plugins, $plugin );
-			$action_type = 'enable-autoupdate=true';
-		}
-		update_site_option( 'wp_auto_update_plugins', $wp_auto_update_plugins );
-		wp_redirect( self_admin_url( "plugins.php?$action_type&plugin_status=$status&paged=$page&s=$s" ) );
+	if ( empty( $plugin ) ) {
+		wp_redirect( self_admin_url( "plugins.php?plugin_status=$status&paged=$page&s=$s" ) );
 		exit;
 	}
+
+	$wp_auto_update_plugins = get_site_option( 'wp_auto_update_plugins', array() );
+
+	if ( in_array( $plugin, $wp_auto_update_plugins, true ) ) {
+		$wp_auto_update_plugins = array_diff( $wp_auto_update_plugins, array( $plugin ) );
+		$action_type            = 'disable-auto-update=true';
+	} else {
+		$wp_auto_update_plugins[] = $plugin;
+		$wp_auto_update_plugins   = array_unique( $wp_auto_update_plugins );
+		$action_type = 'enable-auto-update=true';
+	}
```

Comment:
This doesn't sound right. The "user action" (enable or disable) should always be explicit. Toggling the setting may result in unwanted changes, edge cases, etc. Example: the user opens two tabs and clicks "enable" on both. Will the end result be "enabled" or "disabled"?

Reply:
OK, let me do another push that restores the old code for that part.  Give me just a few minutes